zng/recorditer.go

```diff
@@ -0,0 +1,104 @@
+package zng
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/brimsec/zq/zcode"
+)
+
+var (
+	ErrExhausted = errors.New("called Next() on iterator after last record")
+	ErrMismatch  = errors.New("mismatch between record type and value")
+)
+
+type RecordIter interface {
```

Comment:
Why make this an interface?

Reply:
no good reason, left over from early hacking.  removed.